Universal/Sniffs/DeclareStatements/DeclareStatementsStyleSniff.php

```diff
@@ -0,0 +1,241 @@
+ * 2. Applied to a limited scope using curly braces (the exception to this rule is the `strict_types` directive)
+ * 3. Applied to a limited scope using alternative control structure syntax
+ *
+ * You can also have multiple directives written inside the `declare` directive.
+ * This sniff will check the style of the `declare` directive - by default applied to all directives,
+ * written without braces.
+ *
+ * You can modify the sniff by changing the `declareStyle` option to require or disallow braces,
+ * and require or allow the style to only one of the two directives (`ticks` and `encoding`).
+ *
+ * @since 1.0.0
+ */
+class DeclareStatementsStyleSniff implements Sniff
+{
+
+    /**
+     * Whether to require curly brace style or disallow it
+     * when using declare statements.
+     *
+     * Possible values: requireBraces, disallowBraces.
+     *
+     * Default is disallowBraces.
+     *
+     * @var string
+     */
+    public $declareStyle = 'disallowBraces';
+
+    /**
+     * Which declare to enforce the brace style for.
+     * If not explicitly defined, the brace style will be applied
+     * to all declare directive.
+     *
+     * The exception is the strict_types directive which cannot be
+     * written using curly braces.
+     *
+     * Possible values: ticks, encoding.
+     *
+     * @var array
+     */
+    public $directiveType = [];
+
+    /**
+     * Allowed declare directives.
+     *
+     * @var array
+     */
+    private $allowedDirectives = [
+        'strict_types',
+        'ticks',
+        'encoding',
+    ];
+
+    /**
+     * Returns an array of tokens this test wants to listen for.
+     *
+     * @since 1.0.0
+     *
+     * @return array
+     */
+    public function register()
+    {
+        return [
+            T_DECLARE
+        ];
+    }
+
+    /**
+     * Processes this test, when one of its tokens is encountered.
+     *
+     * @since 1.0.0
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile The file being scanned.
+     * @param int                         $stackPtr  The position of the current token
+     *                                               in the stack passed in $tokens.
+     *
+     * @return void
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        $tokens = $phpcsFile->getTokens();
+
+        $openParenPtr  = $tokens[$stackPtr]['parenthesis_opener'];
+        $closeParenPtr = $tokens[$stackPtr]['parenthesis_closer'];
+
+        if (isset($openParenPtr, $closeParenPtr) === false) {
+            // Parse error or live coding, bow out.
+            return;
+        }
+
+        $directiveStrings = [];
+        // Get the next string and check if it's an allowed directive.
+        // Find all the directive strings inside the declare statement.
+        for ($i = $openParenPtr; $i <= $closeParenPtr; $i++) {
+            if ($tokens[$i]['code'] === \T_STRING) {
+                $directiveStrings[] = $tokens[$i]['content'];
+            }
```

Comment:
```suggestion
            if ($tokens[$i]['code'] === \T_STRING
                && isset($this->allowedDirectives[\strtolower($tokens[$i]['content']))
            ) {
                $directiveStrings[] = $tokens[$i]['content'];
            }
```

(something along the lines of...)

Take note of the `strtolower()` and this may need one or two tests as well.